src/verify_host.jl

```diff
@@ -80,8 +80,9 @@ end
 const MATCH_ANY_RE = r""
 const MATCH_NONE_RE = r"$.^"
 
+# TODO: What to do when `env_host_pattern_regex` returns `nothing`?
```

Comment:
Would the following be a reasonable thing to do and also fix the JET issue?
```diff
--- a/verify_host.jl
+++ b/verify_host.jl
@@ -96,8 +95,8 @@ function env_host_pattern_regex(var::AbstractString)
             delete!(ENV_HOST_PATTERN_CACHE, var)
             return MATCH_NONE_RE
         end
-        old_value, regex = get(ENV_HOST_PATTERN_CACHE, var, (nothing, nothing))
-        old_value == value && return regex
+        old_value, regex = get(ENV_HOST_PATTERN_CACHE, var, (nothing, MATCH_NONE_RE))
+        old_value === nothing && return regex
         regex = host_pattern_regex(value, var)
         ENV_HOST_PATTERN_CACHE[var] = (value, regex)
         return regex
```
Basically make it so that the function always returns a `Regex` by reusing the sentinel value used elsewhere.

Reply:
I like @KristofferC's fix better because having an assertion failure when the unexpected happens seems better than returning some sentinel regex.